File: src/main/java/com/example/repository/OrderStatusRepository.java
package com.example.repository;

import com.example.entity.OrderStatus;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import java.util.Optional;
import java.util.UUID;

@Repository
public interface OrderStatusRepository extends JpaRepository<OrderStatus, UUID> {
    Optional<OrderStatus> findByName(String name);
}

File: src/main/java/com/example/repository/OrderStatusRepository.kt
package com.example.repository

import com.example.entity.OrderStatus
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.redis.core.RedisHash
import org.springframework.stereotype.Repository
import java.util.*

@Repository
@RedisHash
interface OrderStatusRepository: JpaRepository<OrderStatus, UUID> {
    fun findByName(name: String): Optional<OrderStatus>
}